src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -782,15 +781,15 @@ InstanceKlass* SystemDictionary::resolve_instance_class_or_null(Symbol* name,
           if (oldprobe->check_seen_thread(THREAD, PlaceholderTable::LOAD_INSTANCE)) {
             throw_circularity_error = true;
           } else {
-            // case 1: traditional: should never see load_in_progress.
+            // case 3: traditional: should never see load_in_progress.
             while (!class_has_been_loaded && oldprobe != NULL && oldprobe->instance_load_in_progress()) {
 
-              // case 3: bootstrap classloader: prevent futile classloading,
+              // case 1: bootstrap classloader: prevent futile classloading,
               // wait on first requestor
               if (class_loader.is_null()) {
                 SystemDictionary_lock->wait();
               } else {
-              // case 2: traditional with broken classloader lock. wait on first
+              // case 3: traditional with broken classloader lock. wait on first
```

Comment:
Why combine these into case 3 instead of keeping two distinct cases as per the original comment?

Reply:
I could do this.  I could add case 4 for the the broken class_loader lock.  or case 3.1.

Reply:
I went with 4.  The comment now reads as I understand the code.